This patch is a fix for bridge fdb show filter on
bridge device

'bridge fdb show dev br0' after this patch will show
only fdb entries with bridge br0 as dev

Signed-off-by: Roopa Prabhu <roopa@cumulusnetworks.com>

net/core/rtnetlink.c |   11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/net/core/rtnetlink.c b/net/core/rtnetlink.c
index 3d31428..296533f 100644
--- a/net/core/rtnetlink.c
+++ b/net/core/rtnetlink.c
@@ -2421,8 +2421,15 @@ static int rtnl_fdb_dump(struct sk_buff *skb, struct netlink_callback *cb)
 				cops = ops;
 			}
 
-			if (fdev && (fdev->master != br_dev))
-				continue;
+			if (fdev) {
+				if (fdev->priv_flags & IFF_EBRIDGE) {
+					if (fdev != br_dev)
+						continue;
+				} else {
+					if (fdev->master != br_dev)
+						continue;
+				}
+			}
 
 			if (cops && cops->ndo_fdb_dump)
 				idx = cops->ndo_fdb_dump(skb, cb, br_dev, fdev, idx);
